lib/msprime.c

```diff
@@ -2584,9 +2644,13 @@ msp_run_multiple_mergers_coalescent(msp_t *self, double max_time, unsigned long
             re_t_wait = gsl_ran_exponential(self->rng, 1.0 / lambda);
         }
         /* Common ancestors */
-        n = (double) avl_count(&self->populations[0].ancestors);
-        lambda = n * (n - 1.0);
-        ca_t_wait = gsl_ran_exponential(self->rng, 1.0 / lambda);
+        ca_t_wait = DBL_MAX;
+        for (j = 0; j < self->num_populations; j++) {
```

Comment:
Here also, let's get the core functionality working for a single population before we start adding population structure.